src/Core/Grid/Data/Factory/CustomerViewedProductGridDataFactoryDecorator.php

```diff
@@ -0,0 +1,108 @@
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShopBundle\Service\Routing\Router;
+
+/**
+ * Class CustomerViewedProductGridDataFactoryDecorator decorates data from customer viewed products doctrine data factory.
+ */
+final class CustomerViewedProductGridDataFactoryDecorator implements GridDataFactoryInterface
+{
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $customerViewedProductDoctrineGridDataFactory;
+
+    /**
+     * @var Router
+     */
+    private $router;
+
+    /**
+     * @param GridDataFactoryInterface $customerViewedProductDoctrineGridDataFactory
+     * @param Router $router
+     */
+    public function __construct(
+        GridDataFactoryInterface $customerViewedProductDoctrineGridDataFactory,
+        Router $router
+    ) {
+        $this->customerViewedProductDoctrineGridDataFactory = $customerViewedProductDoctrineGridDataFactory;
+        $this->router = $router;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $customerData = $this->customerViewedProductDoctrineGridDataFactory->getData($searchCriteria);
+
+        $records = $this->applyModifications($customerData->getRecords());
+
+        return new GridData(
+            $records,
+            $customerData->getRecordsTotal(),
+            $customerData->getQuery()
+        );
+    }
+
+    /**
+     * @param RecordCollectionInterface $records
+     *
+     * @return RecordCollection
+     */
+    private function applyModifications(RecordCollectionInterface $records)
+    {
+        $modifiedRecord = [];
+
+        foreach ($records as $r) {
+            // Set link to the product in frontoffice
+            $productUrl = $this->router->generate('admin_products_v2_preview', [
+                'productId' => (int) $r['id_product'],
+                'shopId' => (int) $r['id_shop'],
+            ]);
+            $r['product_name'] = '<a href="' . $productUrl . '">' . $r['product_name'] . '</a>';
```

Comment:
I feel we're doing something wrong: you are injecting into a property called product_name some HTML code.

I feel like we are mixing pears and apples.

Maybe we should rather create a new type of column here https://github.com/PrestaShop/PrestaShop/tree/develop/src/Core/Grid/Column/Type/Common ?

Reply:
ping @zuk3975 what do you think?

Reply:
@matks I am using HtmlColumn for that reason ;-)

Reply:
Oooh there is a LinkColumn and it supports unstyled links like this, I will try it: 🚀

Reply:
Done, used `LinkColumn`.